src/module.c

```diff
@@ -555,6 +559,59 @@ int TSDB_create(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     return REDISMODULE_OK;
 }
 
+int TSDB_alter(RedisModuleCtx *ctx, RedisModuleString **argv, int argc){
+    if (argc < 2)
+        return RedisModule_WrongArity(ctx);
+
+    Series *series;
+    RedisModuleString *keyName = argv[1];
+    long long retentionSecs;
+    long long maxSamplesPerChunk;
+    size_t labelsCount;
+    Label *newLabels;
+
+    if (parseCreateArgs(ctx, argv, argc, &retentionSecs, &maxSamplesPerChunk, &labelsCount, &newLabels) != REDISMODULE_OK) {
+        return REDISMODULE_ERR;
+    }
+
+    RedisModuleKey *key = RedisModule_OpenKey(ctx, keyName, REDISMODULE_READ|REDISMODULE_WRITE);
+    if (RedisModule_ModuleTypeGetType(key) != SeriesType) {
+        RedisModule_CloseKey(key);
+        return RedisModule_ReplyWithError(ctx, REDISMODULE_ERRORMSG_WRONGTYPE);
+    }
+    series = RedisModule_ModuleTypeGetValue(key);
+    if (RMUtil_ArgIndex("RETENTION", argv, argc) > 0) {
+        series->retentionSecs = retentionSecs;
+    }
+
+    if (RMUtil_ArgIndex("CHUNK_SIZE", argv, argc) > 0) {
+        series->maxSamplesPerChunk = maxSamplesPerChunk;
+    }
+
+    if (RMUtil_ArgIndex("LABELS", argv, argc) > 0) {
+        RemoveIndexedMetric(ctx, keyName, series->labels, series->labelsCount);
+        // free current labels
+        if (series->labelsCount > 0) {
+            for (int i = 0; i < series->labelsCount; i++) {
+                RedisModule_FreeString(ctx, series->labels[i].key);
+                RedisModule_FreeString(ctx, series->labels[i].value);
+            }
+            free(series->labels);
+        }
+
+        // set new newLabels
+        series->labels = newLabels;
+        series->labelsCount = labelsCount;
+        IndexMetric(ctx, keyName, series->labels, series->labelsCount);
+    }
+
+    RedisModule_CloseKey(key);
+    RedisModule_Log(ctx, "info", "altered existing series");
```

Comment:
I know that its seems weird, but logging per cmd is not something we do in redis.